### ISSUE TITLE
Nightmare process won't exit if page has confirm dialog on page exit

### ISSUE THREAD

**Issue by Fauntleroy** [segmentio/nightmare#824]
When trying to `.end()` a nightmare instance the process will hang forever if there is a `confirm` dialog on page exit. The confirm dialog is also not visible.

Code:

``` js
var Nightmare = require('nightmare');
var nightmare = Nightmare({ show: true });

nightmare
  .goto('https://docs.google.com/forms/d/e/1FAIpQLSe2SK5Vzy82yB9SjLI5B3zfrR1QEaxyjyRGvVxWp_K66p31ZA/viewform')
  .type('input[name="entry.1921379240"]', 'NodeSchool Oakland')
  .end()
  .then(function () {});
```

My view of the electron window:

![screen shot 2016-10-10 at 6 29 43 pm](https://cloud.githubusercontent.com/assets/507047/19255990/d89529d4-8f17-11e6-8a70-d1f84d72fd8f.png)

Is there some way to forcefully close the nightmare instance?

**Comment by Fauntleroy** [comment]
I have isolated the issue. It seems that Google Forms is setting the `unload` event and preventing navigation with the special `unload` confirm dialog.

I have a temporary fix right now:
- Hijack `window.addEventListener` and silently kill attempts to bind `unload` events
- Make `window.onunload` and `window.onbeforeunload` unwritable properties

I'll probably submit a pull req after I've cleaned up my solution.

**Comment by rosshinkley** [comment]
For what it's worth, you could instead prop up a custom preload to handle the `confirm` case manually.  By default, the confirm handler returns the default response, which in this case is _probably_ to stay on the page.

**Comment by Fauntleroy** [comment]
@rosshinkley in this case, `window.confirm` isn't the issue, it's the `unload` and `onbeforeunload` handlers. I managed to fix it with a custom `preload.js` script, which I've submitted in a pull request here: https://github.com/segmentio/nightmare/pull/825

**Comment by rosshinkley** [comment]
Fixed with #825.  Closing.
